res/controllers/DJ TechTools-MIDI Fighter Spectra-scripts.js

```diff
@@ -224,45 +231,36 @@ var MidiFighterSpectra;
                 });
             }
 
+            this.selectDeck = [];
             this.activeDeck = new Deck();
 
-            // Since this is a radio button set, just have one button that
-            // handles setting and unsetting all the LEDs.
-            this.selectDeck = new components.Button({
-                group: "[Channel1]",
-                key: "end_of_track",
-                midi: [0x92, [0x34, 0x54]],
-                input: function(_channel, control, value, _status, group) {
-                    MidiFighterSpectra.controller.activeDeck.setCurrentDeck(group);
-                    for (let i = 0; i < 4; i++) {
-                        for (const midino of this.midi[1]) {
-                            midi.sendShortMsg(this.midi[0], midino + i, (control === midino + i) ? this.on : this.off);
-                        }
-                    }
-                },
-                output: function(value, group, _control) {
-                    // Pulse the deck select button if the track is ending.
-                    const deckNum = parseInt(script.channelRegEx.exec(group)[1]);
-                    for (const midino of this.midi[1]) {
-                        if (value) {
-                            midi.sendShortMsg(this.midi[0] + 1, midino + deckNum - 1, 47);
-                        } else {
-                            midi.sendShortMsg(this.midi[0] - 0xF, midino + deckNum - 1, 33);
-                        }
-                    }
-                },
-                on: engine.getSetting("deckSelectedColor"),
-                off: engine.getSetting("deckUnselectedColor"),
-                connect: function() {
-                    if (engine.getSetting("pulseDeckSelect")) {
-                        for (let i = 0; i < 4; i++) {
-                            this.connections[i] = engine.makeConnection(`[Channel${i + 1}]`, "end_of_track", this.output.bind(this));
+            for (let i = 0; i < 8; i++) {
+                this.selectDeck[i] = new components.Button({
```

Comment:
the button indicies 4-8 don't make much sense to me. Can you elaborate what their purpose is?

Reply:
There are 4 of these buttons across two layers, each group of 4 is a set of radio buttons where only one can be illuminated at a time (indicating which deck is selected). I just put them in a slice to have an easy way to reference them from the XML, I could equally name them "layer1Deck1Button" and "layer2Deck1Button" or something, but the loop made it easier to construct them.